### PR TITLE
[MT-123]: Add basic hash functions

### PR DESCRIPTION
**Pull request by PixoDev**
Add basic hash functions with tests and JUnit setup

- SHA256
- SHA512
- RIPEMD160
- PBKDF2 with SHA512

Also this PR adds Gradle Spongy Castle dependency

### CODE REVIEW

File: android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.spongycastle.jcajce.provider.digest.RIPEMD160
+import java.security.MessageDigest
+
+class Hash {
+
+  /**
+   * Given an array of bytes, generates a SHA256 Hash in bytes
+   */
+  public fun sha256(input: ByteArray): ByteArray {
+    val digest = MessageDigest.getInstance("SHA-256")
```
Comment (sg-gs) [inline android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt]: `hasher` could be a better naming

File: android/src/test/java/com/internxt/mobilesdk/services/crypto/HashTest.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+internal class HashTest {
+  private val SUT: Hash = Hash()
+
+    @Test
+    fun `Should create a SHA256 hash for "test" string correctly`() {
+      // SHA256 "test" string in bytes
+      val expectedValue = byteArrayOf(-97, -122, -48, -127, -120, 76, 125, 101, -102, 47, -22, -96, -59, 90, -48, 21, -93, -65, 79, 27, 43, 11, -126, 44, -47, 93, 108, 21, -80, -16, 10, 8)
+
+      val result = SUT.sha256("test".toByteArray());
+      println(result.contentToString())
```
Comment (sg-gs) [inline android/src/test/java/com/internxt/mobilesdk/services/crypto/HashTest.kt]: remove `println`

File: android/src/test/java/com/internxt/mobilesdk/services/crypto/KeyDerivationTest.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  @Test
+  fun `Should generate a pbkdf2 derivated key correctly`() {
+    val expectedValue = byteArrayOf(-111, -66, 35, 86, 79, 9, -4, -123, 92, -126, -50, -124, -94, 35, -21, -25, -42, 61, -117, 73, -42, -109, 114, 89, 58, 13, -98, -45, -98, 20, 60, -125, -31, -85, 47, 114, 42, 93, -37, -106, -97, -18, -4, -120, 64, 63, 126, 42, -2, 26, -5, -117, 47, 14, 107, 32, -83, -48, -5, 123, 40, 54, -120, 7)
+
+    val password = "password"
+    val salt = "salt"
+    val rounds = 2048
+    val derivedKeyLength = 64
+
+    val derivedKey = SUT.pbkdf2(password, salt, rounds, derivedKeyLength)
```
Comment (sg-gs) [inline android/src/test/java/com/internxt/mobilesdk/services/crypto/KeyDerivationTest.kt:19]: Is this function translated to native to perform this: https://github.com/internxt/drive-web/blob/a3ab49022c03cad73e90820fc8245c846ea84f3b/src/app/crypto/services/utils.ts#L14?

In that case, the test could be better using exactly the same parameters

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: android/src/test/java/com/internxt/mobilesdk/services/crypto/KeyDerivationTest.kt (hunk 3 above)
Comment (PixoDev) [reply android/src/test/java/com/internxt/mobilesdk/services/crypto/KeyDerivationTest.kt:19]: Correct, is that function, will modify the parameters

File: android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt (hunk 1 above)
Comment (PixoDev) [reply android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt]: you mean for the file name? or the function name? sha256Hasher?

Comment (sg-gs) [reply android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt]: for the val name:

`val hasher`

As is the hasher (who hashes) who digests the content

Comment (PixoDev) [reply android/src/main/java/com/internxt/mobilesdk/services/crypto/Hash.kt]: got it

**Pull request merged**